resources/bundles/org.eclipse.core.filesystem/src/org/eclipse/core/internal/filesystem/local/nio/DefaultHandler.java

```diff
@@ -35,13 +37,12 @@ public class DefaultHandler extends NativeHandler {
 
 	@Override
 	public FileInfo fetchFileInfo(String fileName) {
-		Path path = Paths.get(fileName);
+		Path path = Path.of(fileName);
 		FileInfo info = new FileInfo();
-		boolean exists = Files.exists(path);
-		info.setExists(exists);
 
 		try {
 			BasicFileAttributes readAttributes = Files.readAttributes(path, BasicFileAttributes.class);
+			info.setExists(true);
```

Comment:
please add a junit test if this behaves like before for hidden files as the code contains comments that opening hidden files can lead to FileNotFoundException. See uses of org.eclipse.core.filesystem.EFS.ATTRIBUTE_HIDDEN

Reply:
you can look at https://github.com/eclipse-platform/eclipse.platform/pull/1444/files#diff-81300c5bdd85b0fca987936ae34590fbf4bbf835119e1efd68698461d230b817R252 how to create a hidden file. 
But to me it is not clear how/if the Defaulthandler is tested in CI